src/main/java/com/datalogics/pdf/hsm/samples/HsmSignDocument.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2016 Datalogics Inc.
+ */
+
+package com.datalogics.pdf.hsm.samples;
+
+import com.adobe.internal.io.ByteWriter;
+import com.adobe.pdfjt.core.credentials.Credentials;
+import com.adobe.pdfjt.core.exceptions.PDFException;
+import com.adobe.pdfjt.core.exceptions.PDFIOException;
+import com.adobe.pdfjt.core.license.LicenseManager;
+import com.adobe.pdfjt.core.securityframework.CryptoMode;
+import com.adobe.pdfjt.pdf.document.PDFDocument;
+import com.adobe.pdfjt.services.digsig.SignatureFieldInterface;
+import com.adobe.pdfjt.services.digsig.SignatureManager;
+import com.adobe.pdfjt.services.digsig.SignatureOptions;
+import com.adobe.pdfjt.services.digsig.UserInfo;
+import com.adobe.pdfjt.services.digsig.cryptoprovider.JCEProvider;
+import com.adobe.pdfjt.services.digsig.spi.CryptoContext;
+
+import com.datalogics.pdf.hsm.samples.HsmManagerFactory.HsmType;
+import com.datalogics.pdf.hsm.samples.util.DocumentUtils;
+import com.datalogics.pdf.hsm.samples.util.IoUtils;
+import com.datalogics.pdf.security.HsmManager;
+
+import java.io.File;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This is a sample that demonstrates how to sign fields in a document using a Hardware Security Module (HSM). It will
+ * find a specific signature field in a document so that API users can sign the correct field. Note that the password,
+ * private key label, certificate label, and other parameters all must be customized for a particular HSM installation.
+ * This sample is set up to work with a Luna SA HSM device, with an evaluation certificate installed on the default
+ * partition; this HSM is not publicly accessible.
+ */
+public final class HsmSignDocument {
+    private static final Logger LOGGER = Logger.getLogger(HsmSignDocument.class.getName());
+
+    private static final String PASSWORD = ""; // <-- Replace this with the HSM partition password
```

Comment:
We should add functionality so that this sample go out and fetches the password from some file in the directory path.